src/TokenParser/FilterTokenParser.php

```diff
@@ -30,21 +30,24 @@ final class FilterTokenParser extends AbstractTokenParser
 {
     public function parse(Token $token)
     {
-        @trigger_error('The "filter" tag is deprecated since Twig 2.9, use the "apply" tag instead.', E_USER_DEPRECATED);
+        $stream = $this->parser->getStream();
+        $lineno = $token->getLine();
+
+        @trigger_error(sprintf('The "filter" tag is deprecated since Twig 2.9, use the "apply" tag instead. ("%s" at line %d).', $stream->getSourceContext()->getName(), $lineno), E_USER_DEPRECATED);
```

Comment:
I would use `tag instead in "%s" at line %d.` to be more similar to what we're doing with exceptions.

Reply:
Or perhaps like you did elsewhere here: `The "filter" tag in "%s" at line %d is deprecated ...`

Reply:
done.